etcdserver/raft.go

```diff
@@ -226,6 +226,10 @@ func (r *raftNode) start(s *EtcdServer) {
 				if err := r.storage.Save(rd.HardState, rd.Entries); err != nil {
 					plog.Fatalf("raft save state and entries error: %v", err)
 				}
+				if !raft.IsEmptyHardState(rd.HardState) {
+					proposalsCommitted.Set(float64(rd.HardState.Commit))
```

Comment:
This is fine as long as the value satisfies the counter semantics. In the end, if the diff was negative one couldn't do much about it either.
It's what one basically has to do for all exporters as well.